### PR TITLE
Document limitations of GMT xarray accessors

### PR DESCRIPTION
**Pull request by seisman**
**Description of proposed changes**

See https://github.com/GenericMappingTools/pygmt/issues/499 for context.

**Preview**: https://pygmt-dev--2375.org.readthedocs.build/en/2375/api/generated/pygmt.GMTDataArrayAccessor.html

**Reminders**

- [ ] Run `make format` and `make check` to make sure the code follows the style guide.
- [ ] Add tests for new features or tests that would have caught the bug that you're fixing.
- [ ] Add new public functions/methods/classes to `doc/api/index.rst`.
- [ ] Write detailed docstrings for all functions/methods.
- [ ] If wrapping a new module, open a 'Wrap new GMT module' issue and submit reasonably-sized PRs.
- [ ] If adding new functionality, add an example to docstrings or tutorials.
- [ ] Use underscores (not hyphens) in names of Python files and directories.

**Slash Commands**

You can write slash commands (`/command`) in the first line of a comment to perform
specific operations. Supported slash commands are:

- `/format`: automatically format and lint the code
- `/test-gmt-dev`: run full tests on the latest GMT development version

### CODE REVIEW

**Comment by seisman** [comment]
Ping @GenericMappingTools/pygmt-maintainers for reviews.

File: pygmt/accessors.py [hunk 1]
```diff
@@ -31,7 +52,10 @@ class GMTDataArrayAccessor:
     >>> grid.gmt.gtype
     1
 
-    You can also set the GMT specific properties for grids created by yourself:
+    For :class:`xarray.DataArray` grids created by yourself, grid properties
+    ``registration`` and ``gtype`` default to 0 (i.e., a Cartesian
+    gridline-registrated grid). You need to set the correct properties before
```
Comment (weiji14) [inline pygmt/accessors.py:57]: Small nitpick on the order, having registration first, and then the gtype.
```suggestion
    ``registration`` and ``gtype`` default to 0 (i.e., a gridline-registered,
    Cartesian grid). You need to set the correct properties before
```

File: pygmt/accessors.py [hunk 2]
```diff
@@ -45,9 +69,58 @@ class GMTDataArrayAccessor:
...
+    Note that the accessors are created once per :class:`xarray.DataArray`
+    instance, so you may lose these GMT-specific properties after manipulating
+    your grid.
+
+    Inplace assignment operators like ``*=`` don't create new instances, so the
+    properties are still kept:
+
+    >>> grid *= 2.0
+    >>> grid.gmt.registration, grid.gmt.gtype
+    (0, 1)
```
Comment (weiji14) [inline pygmt/accessors.py:90]: Interesting, good to know that in-place assignment preserves the registration and gtype information!

File: pygmt/accessors.py [hunk 3]
```diff
@@ -11,15 +11,36 @@
 @xr.register_dataarray_accessor("gmt")
 class GMTDataArrayAccessor:
     """
-    GMT extension for :class:`xarray.DataArray`.
+    GMT accessor for :class:`xarray.DataArray`.
 
-    The extension provides easy ways to access and change the GMT specific
-    properties about grids. Currently, two properties are available:
+    The accessor extends :class:`xarray.DataArray` to store GMT-specific
+    properties about grids, which are important for PyGMT to correctly process
```
Comment (weiji14) [inline pygmt/accessors.py:17]: Technically xarray accessors are not just about setting properties/attributes. You may need to change this sentence if the `.gmt.imshow()` function at #2372 is added to this accessor class.

**Review by weiji14 — APPROVED** [review]
Just one very minor suggestion, but otherwise I think it's documented really well!

File: pygmt/accessors.py [hunk 4]
```diff
@@ -11,15 +11,36 @@
...
+
+    Notes
+    -----
+
+    Due to the limitations of xarray accessors, the GMT accessors are created
+    once per :class:`xarray.DataArray` instance. You may lose these
+    GMT-specific properties when manipulating grids (e.g., arithmetic and slice
+    operations) or when accessing a :class:`xarray.DataArray` from a
+    :class:`xarray.Dataset`. In these cases, you need to manually set these
+    properties beforing passing it to PyGMT.
```
Comment (yvonnefroehlich) [inline pygmt/accessors.py:37]:
```suggestion
    Notes
    -----

    Due to the limitations of xarray accessors, the GMT accessors are created
    once per :class:`xarray.DataArray` instance. You may lose these
    GMT-specific properties when manipulating grids (e.g., arithmetic and slice
    operations) or when accessing a :class:`xarray.DataArray` from a
    :class:`xarray.Dataset`. In these cases, you need to manually set these
    properties before passing the grid to PyGMT.
```

File: pygmt/accessors.py (hunk 3 above)
Comment (seisman) [reply pygmt/accessors.py:17]: Will rephrase this sentence in PR #2372.

**Review by michaelgrund — APPROVED** [review]

**Pull request merged**